reactExercise/solutions/zihe/ex5/src/Lifecycle.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+class Lifecycle extends React.Component {
+    constructor() {
+        super()
+        this.state = {
+            x1: false,
+            x2: false,
+            x3: false,
+            loading: true,
+        }
+    }
+    componentDidMount() {
+        setTimeout(() => {
+            this.setState({ loading: false })
+        }, 3000)
+    }
+    shouldComponentUpdate(nextProps, nextState) {
+        if (nextState.x3 === true && nextState.x1 === false && nextState.x2 === false) {
```

Comment:
如果是boolean type的话, 我们可以直接evaluate 当然如果你觉得 === true和 === false 更清晰的话 这样也可以

```suggestion
        if (nextState.x3 && !nextState.x1 && !nextState.x2) {
```